src/logger/BaseLogger_test.go

```diff
@@ -0,0 +1,228 @@
+
+func TestSubmitsToDemoUrlViaHttp(t *testing.T) {
+	helper := NewTestHelper()
+	logger := NewBaseLoggerAgentUrl(helper.mockAgent, strings.Replace(helper.demoURL, "https://", "http://", 1))
+	assert.Equal(t, strings.Index(logger.Url, "http://"), 0)
+	message := [][]string{}
+	message = append(message, []string{"agent", logger.Agent})
+	message = append(message, []string{"version", logger.Version})
+	message = append(message, []string{"now", string(helper.mockNow)})
+	message = append(message, []string{"protocol", "http"})
+	//TODO: toimplement stringify method
+	msg := "Json.stringify(message)"
+	//TODO: implement parsable method
+	assert.True(t, "parsable(msg)")
+	logger.Submit(msg)
+	assert.Equal(t, logger.SubmitFailures, 0)
+	assert.Equal(t, logger.SubmitSuccesses, 1)
+}
+
+func TestSubmitsToDemoUrlWihoutCompression(t *testing.T) {
+	helper := NewTestHelper()
+	logger := NewBaseLoggerAgentUrl(helper.mockAgent, helper.demoURL)
+	logger.SkipCompression = true
+	assert.True(t, logger.SkipCompression)
+	message := [][]string{}
+	message = append(message, []string{"agent", logger.Agent})
+	message = append(message, []string{"version", logger.Version})
+	message = append(message, []string{"now", string(helper.mockNow)})
+	message = append(message, []string{"protocol", "https"})
+	message = append(message, []string{"skip_compression", "true"})
+	//TODO: toimplement stringify method
+	msg := "Json.stringify(message)"
+	//TODO: implement parsable method
+	assert.True(t, "parsable(msg)")
+	logger.Submit(msg)
+	assert.Equal(t, logger.SubmitFailures, 0)
+	assert.Equal(t, logger.SubmitSuccesses, 1)
+}
+
+func TestSubmitsToDeniedUrl(t *testing.T) {
+	helper := NewTestHelper()
+	for _, url := range helper.mockURLSinvalid {
+		logger := NewBaseLoggerAgentUrl(helper.mockAgent, url)
+		assert.True(t, logger.Enableable)
+		assert.True(t, logger.Enabled)
+		logger.Submit("{}")
+		assert.Equal(t, logger.SubmitFailures, 1)
+		assert.Equal(t, logger.SubmitSuccesses, 0)
+	}
+}
+
+func TestSubmitsToQueue(t *testing.T) {
+	helper := NewTestHelper()
+	queue := []string{}
+	logger := NewBaseLoggerAgentQueue(helper.mockAgent, queue)
+	assert.Equal(t, logger.Queue, queue)
+	assert.Nil(t, logger.Url)
+	assert.True(t, logger.Enableable)
+	assert.True(t, logger.Enabled)
+	assert.Equal(t, len(queue), 0)
+	logger.Submit("{}")
+	assert.Equal(t, len(queue), 1)
+	logger.Submit("{}")
+	assert.Equal(t, len(queue), 2)
+	assert.Equal(logger.SubmitFailures, 0)
+	assert.Equal(logger.SubmitSuccesses, 0)
+}
+
+func TestUsesSkipOptions(t *testing.T) {
+	helper := NewTestHelper()
+	logger := NewBaseLoggerAgentUrl(helper.mockAgent, helper.demoURL)
+	assert.False(t, logger.SkipCompression)
+	assert.False(t, logger.SkipSubmission)
+
+	logger.SkipCompression = true
+	assert.True(t, logger.SkipCompression)
+	assert.False(t, logger.SkipSubmission)
+
+	logger.SkipCompression = false
+	logger.SkipSubmission = true
+	assert.False(t, logger.SkipCompression)
+	assert.True(t, logger.SkipSubmission)
+}
```

Comment:
You are missing a lot of getters and setters for the logger instances specifically. In some ways, not a big deal but we also want to test the getters and setters to make sure they are working.